src/ucp/core/ucp_am.c

```diff
@@ -92,30 +94,28 @@ static ucs_status_t ucp_worker_set_am_handler_common(ucp_worker_h worker,
                                                      uint16_t id)
 {
     size_t num_entries;
-    ucp_am_entry_t *am_cbs;
-    int i;
+    ucs_status_t status;
+    unsigned i, capacity;
 
     UCP_CONTEXT_CHECK_FEATURE_FLAGS(worker->context, UCP_FEATURE_AM,
                                     return UCS_ERR_INVALID_PARAM);
 
-    if (id >= worker->am.cbs_array_len) {
+    if (id >= ucs_array_length(&worker->am)) {
         num_entries = ucs_align_up_pow2(id + 1, UCP_AM_CB_BLOCK_SIZE);
-        am_cbs      = ucs_realloc(worker->am.cbs, num_entries *
-                                  sizeof(ucp_am_entry_t),
-                                  "UCP AM callback array");
-        if (ucs_unlikely(am_cbs == NULL)) {
-            ucs_error("failed to grow UCP am cbs array to %zu", num_entries);
-            return UCS_ERR_NO_MEMORY;
+        status = ucs_array_reserve(ucp_am_cbs, &worker->am, num_entries);
```

Comment:
maybe we don't need num_entries, just
` status = ucs_array_reserve(ucp_am_cbs, &worker->am, id + 1);`